src/Appwrite/Platform/Modules/Databases/Http/Databases/Collections/Attributes/Action.php

```diff
@@ -366,13 +366,20 @@ protected function createAttribute(string $databaseId, string $collectionId, Doc
                 'filters' => $filters,
                 'options' => $options,
             ]);
-
+            if (!$dbForProject->getAdapter()->getSupportForSpatialIndexNull() && in_array($attribute->getAttribute('type'), Database::SPATIAL_TYPES) && $attribute->getAttribute('required')) {
```

Comment:
@abnegate Let's make an adapter call here, which will check all scenarios before creating an attribute.
And test it in Utopia